pytorch_lightning/strategies/launchers/ddp_spawn.py

```diff
@@ -0,0 +1,164 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from collections import UserList
+from multiprocessing.queues import SimpleQueue
+from typing import Any, Callable, NamedTuple, Optional
+
+import numpy as np
+import torch
+import torch.multiprocessing as mp
+
+import pytorch_lightning as pl
+from pytorch_lightning.strategies.launchers.base import Launcher
+from pytorch_lightning.strategies.strategy import Strategy
+from pytorch_lightning.trainer.states import TrainerFn, TrainerState
+from pytorch_lightning.utilities.apply_func import apply_to_collection, move_data_to_device
+from pytorch_lightning.utilities.distributed import rank_zero_debug
+from pytorch_lightning.utilities.model_helpers import is_overridden
+from pytorch_lightning.utilities.types import _PATH
+
+
+class DDPSpawnLauncher(Launcher):
```

Comment:
naming wise, this has nothing to do with DDP/DistributedDataParallel. It's a general torch multi-processing spawn launcher

Reply:
yeah! will update this... did this initially to check how many launchers we actually needed so kept that prefix.